src/pages/roadmap.md

```diff
@@ -1,9 +1,215 @@
----
-hide_table_of_contents: true
----
+# The future of XMTP
 
-# Roadmap for XMTP
+This roadmap outlines plans for the future of these key areas of XMTP:
 
-We are working on an updated roadmap. Please stay tuned.
+- Identity
+- Consent
+- Decentralization
 
-In the meantime, please head to the [XMTP Community Forums](https://community.xmtp.org/) if you have any questions about what's in store for XMTP development.
\ No newline at end of file
+## The future of identity
+
+In 2025, wallet associations will transition to the XMTP appchain, providing trustless, censorship-resistant infrastructure for maintaining consistent state, while preserving the privacy, security, and unlinkability guarantees of the system.
+
+Users can continue to optionally link their wallets when they choose to do so using the `IdentityUpdates` contract:
+
+```solidity
+event IdentityUpdateCreated(
+    bytes32 inboxId,
+    bytes update,      // Signed IdentityAction from SDK
+    uint64 sequenceId
+);
+```
+
+SDKs will maintain the same methods for working with messaging identities, allowing identity association history to move onchain without affecting existing implementations. 
+
+This move onchain will make this history permissionlessly accessible and anyone will be able to construct a list of wallets associated with an identity. Users who wish to maintain a private, unlinkable identity should use a separate identity for any public wallet addresses. 
+
+Applications will have two paths for working with messaging identities:
+
+1. **Using XMTP SDKs:** Developers using XMTP for messaging can continue leveraging the SDKs’ built-in identity management and third-party libraries for name resolution.
+2. **Using the Appchain:** Developers not directly providing messaging, such as infrastructure and data providers, can independently monitor identity updates by:
+    - Watching chain events and building state tracking
+    - Using third-party APIs to retrieve the latest state
+    - Accessing state permissionlessly via community-built subgraphs
+
+The transition to the XMTP appchain will be seamless for users and developers. Existing identities will automatically migrate without disrupting functionality. The SDK interface will remain unchanged, and detailed migration guides will be provided to ensure smooth adoption. 
+
+When the system transitions to the XMTP appchain, rate limiting for identity operations will be managed through gas fees for each operation. Identity operations will incur small gas fees, likely fractions of a cent. To ease user onboarding, applications can register onchain as fee payers, covering these costs for new users during sign-up.
+
+XMTP is also planning to update the protocol to allow you to temporarily silence notifications (mute conversations) while maintaining consent preferences.
```

Comment:
Have we committed to doing this at the protocol level. I thought it was going to happen at the Convos level, but maybe I wasn't in the meeting